uint32 size
NeighbourId[] list